libs/agno/agno/agent/agent.py

```diff
@@ -4146,12 +4183,19 @@ def search_knowledge_base(self, query: str) -> str:
         Returns:
             str: A string containing the response from the knowledge base.
         """
+        # Get knowledge filters from context if available
+        filters = None
+        if self.context and "knowledge_filters" in self.context:
+            filters = self.context["knowledge_filters"]
+            from agno.utils.log import log_debug
 
-        # Get the relevant documents from the knowledge base
+            log_debug(f"Using knowledge filters from context: {filters}")
+
+        # Get the relevant documents from the knowledge base, passing filters
         self.run_response = cast(RunResponse, self.run_response)
         retrieval_timer = Timer()
         retrieval_timer.start()
-        docs_from_knowledge = self.get_relevant_docs_from_knowledge(query=query)
+        docs_from_knowledge = self.get_relevant_docs_from_knowledge(query=query, filters=filters)
```

Comment:
The filters should also automatically apply to the agent's search_knowledge functions

Reply:
done, pushing the change